### PR TITLE
Post JSON data for non-clinic lines

### PR DESCRIPTION
**Pull request by imsickofmaps**
Format as per: https://praekelt.atlassian.net/browse/NDOH-16

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@smn ready for review. As per discussion success/error handling will be in issue #32

File: go-app-chw.js [hunk 1]
```diff
@@ -391,6 +412,32 @@ go.utils = {
...
+    jembi_json_api_call: function (json_doc, im) {
+        var http = new JsonApi(im, {
+          auth: {
+            username: im.config.jembi.username,
+            password: im.config.jembi.password
+          }
+        });
+        return http.post(im.config.jembi.url_json, {
+          data: json_doc
+        });
```
Comment (smn) [inline go-app-chw.js:427]: Is this worth the extra method? Perhaps better at this point to create `self.http` in `init` and just use `self.http.post(...)` where you need it from there?

File: go-app-chw.js [hunk 2]
```diff
@@ -921,7 +968,23 @@ go.app = function() {
...
                         'to register at her nearest clinic.'),
 
-                next: 'states:start'
+                next: 'states:start',
+                events: {
+                    'state:enter': function() {
+                        var built_json = go.utils.build_json_doc(self.contact, self.user, "pre-registration");
+                        return go.utils.jembi_json_api_call(built_json, self.im)
+                            .then(function(result) {
+                                if (result.code >= 200 && result.code < 300){
```
Comment (smn) [inline go-app-chw.js:966]: I'd suggest you fail fast here and not do the logging or always `return true`? Your fixture will let you test whatever you want.

File: go-app-personal.js [hunk 3]
```diff
@@ -391,6 +412,32 @@ go.utils = {
...
+            password: im.config.jembi.password
+          }
+        });
+        return http.post(im.config.jembi.url_json, {
+          data: json_doc
+        });
+    },
+
+    get_JSON_template: function() {
+      var JSON_template = {
```
Comment (smn) [inline go-app-personal.js]: Is the split between `get_JSON_template` and `build_json_doc` worth it? Seems like unnecessary overhead.

File: test/personal.test.js [hunk 4]
```diff
@@ -43,14 +43,15 @@ describe("app", function() {
...
-                .setup(function(api) {
-                    api.contacts.add( {
-                        msisdn: '+27001'
-                    });
-                })
+                // .setup(function(api) {
+                //     api.contacts.add( {
+                //         msisdn: '+27001'
+                //     });
+                // })
```
Comment (smn) [inline test/personal.test.js]: remove?

**Comment by smn** [comment]
minor stuff, looks good otherwise.

File: go-app-chw.js (hunk 1 above)
Comment (imsickofmaps) [inline go-app-chw.js:427]: I was going to do that before I realised one is JSON the other is not

File: go-app-chw.js (hunk 2 above)
Comment (imsickofmaps) [inline go-app-chw.js:966]: We need to capture a metric for the dashboard over in #32

**Comment by imsickofmaps** [comment]
Is that a :+1:?

**Comment by imsickofmaps** [comment]
Over the shoulder :+1: from @smn

**Pull request merged**